lib/masamune/schema/map.rb

```diff
@@ -24,6 +24,34 @@
 
 module Masamune::Schema
   class Map
+    class JSONEncoder < SimpleDelegator
+      def initialize(io, store)
+        super io
+        @store = store
+      end
+
+      def gets(*a)
+        line = __getobj__.gets(*a)
+        return unless line
+        return line if skip?
+        encode(line.split(separator)).join(separator)
+      end
+
+      private
+
+      def skip?
+        @store.json_encoding == :quoted
+      end
+
+      def encode(fields = [])
+        fields.map { |field| field =~ /^{|}$/ ? %Q{"#{field.gsub('"', '""')}"} : field }
```

Comment:
if the field looks like raw JSON (e.g. `{"enabled":true}`) then quote it (e.g. `"{""enabled"":true}"`) so that `CSV` can parse the field